### PR TITLE
Feature/treeview

### PR DESCRIPTION
**Pull request by ydomenjoud**

### CODE REVIEW

File: projects/kalidea/kaligraphi/src/lib/atoms/kal-tree/kal-tree.component.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+})
+export class KalTreeComponent<T extends { children?: T[], title: string, level?: number }> implements OnInit {
+  @Input() dataSource: KalTreeDataSource<T>;
+
+  get nodesList() {
+    return this.flatten(this.dataSource.data);
+  }
+
+  drop($event) {
+    console.log($event);
```
Comment (jsp-up) [inline projects/kalidea/kaligraphi/src/lib/atoms/kal-tree/kal-tree.component.ts]: emit event on drop ?

**Review by jsp-up — CHANGES_REQUESTED** [review]

File: projects/kalidea/kaligraphi/src/lib/atoms/kal-tree/kal-tree.component.ts (hunk 1 above)
Comment (ydomenjoud) [reply projects/kalidea/kaligraphi/src/lib/atoms/kal-tree/kal-tree.component.ts]: This PR is  a proof of concept for michael.not a real PR.

**Review by jsp-up — APPROVED** [review]

File: projects/kalidea/kaligraphi/src/lib/atoms/kal-tree/components/kal-tree-node.component.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  @Input() disabled;
+
+  @Input() tabIndex = 0;
+
+  node: T;
+
+  @ContentChildren(KalTreeNodeOutletDirective) nodeOutlet: QueryList<KalTreeNodeOutletDirective>;
+
+  @Input()
+  set kalNestedTreeNode(node) {
```
Comment (MichaelGrosio) [inline projects/kalidea/kaligraphi/src/lib/atoms/kal-tree/components/kal-tree-node.component.ts:36]: type node

File: projects/kalidea/kaligraphi/src/lib/atoms/kal-tree/components/kal-tree-node.component.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    {provide: CdkTreeNode, useExisting: KalTreeNodeComponent}
+  ]
+})
+export class KalTreeNodeComponent<T> extends CdkNestedTreeNode<T> implements AfterContentInit, OnDestroy {
+
+  @Input() disabled;
+
+  @Input() tabIndex = 0;
+
+  node: T;
```
Comment (MichaelGrosio) [inline projects/kalidea/kaligraphi/src/lib/atoms/kal-tree/components/kal-tree-node.component.ts:31]: format code

File: src/app/atoms/atoms.module.ts [hunk 4]
```diff
@@ -15,6 +15,7 @@ import { SliderComponent } from 'src/app/atoms/slider/slider.component';
 import { MenuComponent } from 'src/app/atoms/menu/menu.component';
 import { CardComponent } from 'src/app/atoms/card/card.component';
 import { SnackbarComponent } from './snackbar/snackbar.component';
```
Comment (MichaelGrosio) [inline src/app/atoms/atoms.module.ts:17]: format code

File: projects/kalidea/kaligraphi/src/lib/atoms/kal-tree/classes/kal-tree-control.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+import { NestedTreeControl } from '@angular/cdk/tree';
+import { KalTreeNode } from './kal-tree-node';
+
+export class KalTreeControl extends NestedTreeControl<KalTreeNode> {
```
Comment (MichaelGrosio) [inline projects/kalidea/kaligraphi/src/lib/atoms/kal-tree/classes/kal-tree-control.ts:4]: Why did you make a kalTreeControl, you can use NestedTreeControl !

**Pull request merged**